src/App.tsx

```diff
@@ -15,48 +16,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export enum SortOptions {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  None = '',
+}
+
+const handleSort
+  = (goods: string[],
+    options: { activeSort: SortOptions; isReversed: boolean }) => {
+    const preparedGoods = [...goods];
+
+    preparedGoods.sort((a, b) => {
+      switch (options.activeSort) {
+        case SortOptions.Alphabetically:
+          return a.localeCompare(b);
+
+        case SortOptions.Length:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (options.isReversed) {
+      preparedGoods.reverse();
+    }
+
+    return preparedGoods;
+  };
+
 export const App: React.FC = () => {
+  const [activeSort, setActiveSort]
+    = useState<SortOptions>(SortOptions.None);
+
+  const [isReversed, setIsReversed]
+    = useState(false);
+
+  const visibleGoods = handleSort(goodsFromServer, {
+    activeSort: activeSort || SortOptions,
+    isReversed,
+  } as { activeSort: SortOptions; isReversed: boolean });
+
+  // function setReversed() {
+  //   setIsReversed(!isReversed);
+  // }
+
+  function setSort(selectedOption: SortOptions) {
+    setActiveSort(selectedOption);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': activeSort !== 'alphabetically' })}
+          onClick={() => setSort(SortOptions.Alphabetically)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': activeSort !== 'length' })}
+          onClick={() => setSort(SortOptions.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(activeSort || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSort(SortOptions.None);
+              setIsReversed(false);
+            }}
```

Comment:
```suggestion
            onClick={reset}
```

Reply:
<img width="645" alt="image" src="https://github.com/mate-academy/react_list-of-goods-hooks/assets/70220667/f69ad8d5-374b-4cb3-8675-8b34d2fd66df">